designs/v1alpha4-api.md

```diff
@@ -0,0 +1,130 @@
+# v1alpha4 API Proposal
+This document proposes comprehensive Provisioner API improvements prior to the v0.4 release. Due to minor backwards incompatible changes, this will result in a API version bump (v1alpha3 -> v1alpha4). This is in accordance with [Kubernetes API Deprecation Policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/).
+
+These changes recommend:
+1. Cloud Provider specific extensions under `spec.provider`.
+2. Removal of `spec.cluster`.
+3. Pluralization of `spec.architecture` and `spec.operatingSystem`.
```

Comment:
This was pretty confusing, I thought you meant you were going to change the names to (eg) `operatingSystems`.

Reply:
I do!